netebpfext/net_ebpf_ext_hook_provider.h

```diff
@@ -184,6 +184,21 @@ net_ebpf_extension_hook_invoke_programs(
     _In_ net_ebpf_extension_wfp_filter_context_t* filter_context,
     _Out_ uint32_t* result);
 
+/**
+ * @brief Expand stack and invoke all the eBPF programs attached to the specified filter context.
+ *
+ * @param[in] program_context Context to pass to eBPF program.
+ * @param[in] filter_context Filter context to invoke the programs from.
+ * @param[out] result Return value from the eBPF programs.
+ *
+ * @retval ebpf_result_t Status of the program invocation.
+ */
+ebpf_result_t
+net_ebpf_extension_hook_expand_stack_and_invoke_programs(
+    _Inout_ void* program_context,
+    _In_ net_ebpf_extension_wfp_filter_context_t* filter_context,
```

Comment:
can filter_context be const?  or should it be `_Inout_` if the structure is modified inside the call?

```suggestion
    _In_ const net_ebpf_extension_wfp_filter_context_t* filter_context,
```

Reply:
The function acquires filter context lock, so it should be `_Inout_`

Reply:
Updated.